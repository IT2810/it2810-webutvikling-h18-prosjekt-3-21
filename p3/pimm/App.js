import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { HomeScreen } from "./HomeScreen";
import { EditTaskScreen } from "./EditTaskScreen";

export default class App extends React.Component {
  constructor(props) {
    super(props);
  }
  state = {
    currentTab: "HomeScreen"
  };
  handleAddTask = () => {
    this.setState({ currentTab: "AddTaskScreen" });
  };
  handlePressBack = () => {
    this.setState({ currentTab: "HomeScreen" });
  };
  handleEditTask = (id, desc) => {
    displayedScreen = (
      <EditTaskScreen parent={this} taskid={id} taskdesc={desc} />
    );
    this.setState({ currentTab: "EditTaskScreen" });
  };

  render() {
    if (this.state.currentTab === "HomeScreen") {
      displayedScreen = <HomeScreen parent={this} />;
    } else if (this.state.currentTab === "AddTaskScreen") {
      displayedScreen = <EditTaskScreen parent={this} />;
    }
    return displayedScreen;
  }
}