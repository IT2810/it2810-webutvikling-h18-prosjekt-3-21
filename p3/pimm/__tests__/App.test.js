import React from "react";
import App from "../App";

import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<App />);
  let treeJSON = tree.toJSON();
  expect(treeJSON.type).toEqual("View");
  expect(treeJSON).toMatchSnapshot();

  let inst = tree.root;

  // Test state changes in App.js
  expect(inst.state.currentTab).toEqual("HomeScreen");
  inst.handleEditTask();
  expect(inst.state.currentTab).toEqual("EditTaskScreen");
  inst.handlePressBack();
  expect(inst.state.currentTab).toEqual("HomeScreen");
  inst.handleNavbarPress("Pedometer");
  expect(inst.state.currentTab).toEqual("Pedometer");
});
