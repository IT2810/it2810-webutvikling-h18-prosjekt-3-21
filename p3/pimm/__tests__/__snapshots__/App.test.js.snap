// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightblue",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "padding": 0,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "lightblue",
            "borderBottomWidth": 0,
            "borderColor": "#29436d",
            "borderLeftWidth": 0,
            "borderWidth": 1,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 30,
                "fontWeight": "700",
                "textShadowColor": "#29436d",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 0,
              }
            }
          >
            TO DO
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5176A1",
            "borderColor": "#29436d",
            "borderRadius": 0,
            "borderRightWidth": 0,
            "borderWidth": 1,
            "flex": 1,
            "justifyContent": "center",
            "marginTop": 3,
            "opacity": 1,
            "width": "100%",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#444",
                "fontSize": 29,
                "fontWeight": "600",
                "textShadowColor": "#29436d",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 0,
              }
            }
          >
            COMPLETED
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 7,
          "marginBottom": 20,
          "marginTop": 10,
          "width": "100%",
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5176A1",
          "borderColor": "#29436d",
          "borderRadius": 60,
          "borderWidth": 3,
          "bottom": "5%",
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 0,
          "position": "absolute",
          "right": "5%",
          "width": 60,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 80,
              "fontWeight": "700",
              "height": 78,
              "lineHeight": 80,
              "textShadowColor": "#29436d",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 0,
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "flexDirection": "row",
        "height": "8%",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#f7f7f7",
          "borderTopWidth": 1,
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#555",
              "fontSize": 20,
              "fontWeight": "900",
              "justifyContent": "center",
            },
            Object {
              "color": "#333",
            },
          ]
        }
      >
        TASK LIST
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#aaa",
          "borderTopWidth": 1,
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#555",
              "fontSize": 20,
              "fontWeight": "900",
              "justifyContent": "center",
            },
            "",
          ]
        }
      >
        STEP COUNTER
      </Text>
    </View>
  </View>
</View>
`;
